### PR TITLE
feat: social share

### PR DESCRIPTION
**Pull request by 2fd**
new environment variable `REACT_APP_SHARE_SCENE_URL`:

| environment | `REACT_APP_SHARE_SCENE_URL` |
|----|----|
|`localhost` | `http://localhost:5000/v1/share` |
| `builder.decentraland.today` | `https://share.decentraland.zone/b` |
| `builder.decentraland.zone` | `https://share.decentraland.zone/b` |
| `builder.decentraland.org` | `https://share.decentraland.org/b` |

Please note:
- the texts are not the definitive

### CODE REVIEW

File: src/components/Modals/ShareModal/ShareModal.container.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Dispatch } from 'redux'
+import { connect } from 'react-redux'
+import { RootState } from 'modules/common/types'
+// import { getData as getDeployments } from 'modules/deployment/selectors'
+import { MapStateProps, MapDispatchProps, OwnProps } from './ShareModal.types'
+import DeployModal from './ShareModal'
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.container.ts]: Rename to ShareModal and move to bottom of the import list
As a convention we place the relative imports at the end

File: src/components/Modals/ShareModal/ShareModal.css [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  display: block;
+  background: #393440;
+}
+
+.ShareModal .share-modal .thumbnail {
+  width: calc(480px - 48px - 48px);
+  height: calc(360px - 48px - 48px);
+  background-size: contain;
+  background-position: center center;
+  background: #393440;
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.css]: Can't we derive this color from an existing variable + opacity?

File: src/components/Modals/ShareModal/ShareModal.css [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  font-weight: 100;
+  text-align: center;
+  width: 100%;
+}
+
+.ShareModal .share-modal .thumbnail,
+.ShareModal .share-modal .button {
+  border-radius: 8px;
+  display: block;
+  background: #393440;
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.css]: Can't we derive this color from an existing variable + opacity?

File: src/components/Modals/ShareModal/ShareModal.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
+import * as React from 'react'
+import { Header, Loader, Button } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+// import Icon from 'components/Icon'
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.tsx]:
```suggestion
```

File: src/components/Modals/ShareModal/ShareModal.container.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Dispatch } from 'redux'
+import { connect } from 'react-redux'
+import { RootState } from 'modules/common/types'
+// import { getData as getDeployments } from 'modules/deployment/selectors'
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.container.ts]:
```suggestion
```

File: src/components/Modals/ShareModal/ShareModal.tsx [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+import { Props, State, ShareModalType } from './ShareModal.types'
+import './ShareModal.css'
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.tsx]: Import order

File: src/components/Modals/ShareModal/ShareModal.tsx [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    e.preventDefault()
+    const width = 600
+    const height = 250
+    const top = Math.ceil((window.outerHeight / 2) - (height / 2))
+    const left = Math.ceil((window.outerWidth / 2) - (width / 2))
+    window.open(e.currentTarget.href,'targetWindow',`toolbar=no,location=0,status=no,menubar=no,scrollbars=yes,resizable=yes,width=${width},height=${height},top=${top},left=${left}`)
+    this.props.onShare(target)
+  }
+
+  handleShareWithFacebook = (e: React.MouseEvent<HTMLAnchorElement>) => this.handleShare(e, ShareTarget.FACEBOOK)
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.tsx]: Please add brackets {}

File: src/components/Modals/ShareModal/ShareModal.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  handleShareWithTwitter = (e: React.MouseEvent<HTMLAnchorElement>) => this.handleShare(e, ShareTarget.TWITTER)
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.tsx]: Please add brackets {}

File: src/components/Modals/ShareModal/ShareModal.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    return encodeURI(t('share_modal.uri.facebook'))
+  }
+
+  getTwitterLink = () => {
+    return encodeURI(t('share_modal.uri.twitter'))
+  }
+
+  getShareLink = () => {
+    const { project } = this.props
+    return 'https://share.decentraland.org/scene/' + project.id
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.tsx]: Should we move this URL to a variable?

File: src/components/Modals/ShareModal/index.ts [hunk 10]
```diff
@@ -0,0 +1,3 @@
+import DeployModal from './ShareModal.container'
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/index.ts]: Rename to ShareModal

File: src/components/ShareButton/ShareButton.container.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
+import { connect } from 'react-redux'
+
+import { RootState } from 'modules/common/types'
+import { openModal } from 'modules/modal/actions'
+import { MapStateProps, MapDispatch, MapDispatchProps } from './ShareButton.types'
+import ShareButton from './ShareButton'
```
Comment (belohlavek) [inline src/components/ShareButton/ShareButton.container.ts]: Import order

File: src/components/ShareButton/ShareButton.tsx [hunk 12]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react'
+
+import { Popup } from 'decentraland-ui'
+// import { t, T } from 'decentraland-dapps/dist/modules/translation/utils'
```
Comment (belohlavek) [inline src/components/ShareButton/ShareButton.tsx]:
```suggestion
```

File: src/components/ViewPage/ViewMenu/ViewMenu.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react'
+import { Button } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Props, State } from './ViewMenu.types'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewMenu/ViewMenu.tsx]: Import order

File: src/components/ViewPage/ViewPage.container.ts [hunk 14]
```diff
@@ -0,0 +1,30 @@
+import { connect } from 'react-redux'
+import { getCurrentProject, isFetching } from 'modules/project/selectors'
+import { RootState } from 'modules/common/types'
+import { isPreviewing , isReady } from 'modules/editor/selectors'
+import { MapStateProps, MapDispatch, MapDispatchProps } from './ViewPage.types'
+import ViewPage from './ViewPage'
+import { isLoggedIn } from 'modules/auth/selectors'
+// import { openModal } from 'modules/modal/actions'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.container.ts]:
```suggestion
```

File: src/components/ViewPage/ViewPage.container.ts [hunk 15]
```diff
@@ -0,0 +1,30 @@
+import { connect } from 'react-redux'
+import { getCurrentProject, isFetching } from 'modules/project/selectors'
+import { RootState } from 'modules/common/types'
+import { isPreviewing , isReady } from 'modules/editor/selectors'
+import { MapStateProps, MapDispatch, MapDispatchProps } from './ViewPage.types'
+import ViewPage from './ViewPage'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.container.ts]: Import order

File: src/components/ViewPage/ViewPage.container.ts [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+  isReady: isReady(state),
+  isFetching: isFetching(state) && !isReady(state),
+  isLoggedIn: isLoggedIn(state),
+  currentProject: getCurrentProject(state),
+  currentScene: getCurrentScene(state),
+  currentAuthor: getCurrentAuthor(state)
+})
+
+const mapDispatch = (dispatch: MapDispatch): MapDispatchProps => ({
+  // onOpenModal: name => dispatch(openModal(name)),
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.container.ts]:
```suggestion
```

File: src/components/ViewPage/ViewPage.css [hunk 17]
```diff
@@ -0,0 +1,121 @@
+.ViewPage {
+  display: block;
+  position: relative;
+  background: #242129;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]: Can we derive this color from an existing variable?

File: src/components/ViewPage/ViewPage.css [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+.ViewPage .thumbnail img {
+  width: 100%;
+  border-radius: 12px 12px 0 0;
+  position: relative;
+  height: fit-content;
+}
+
+.ViewPage .thumbnail {
+  height: 600px;
+  background: #393440;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]: Can we derive this color from an existing variable?

File: src/components/ViewPage/ViewPage.css [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+}
+.ViewPage .detail .title h1 {
+  font-size: 21px;
+  font-weight: 500;
+  line-height: 30px;
+  margin: 0;
+}
+
+.ViewPage .detail .author {
+  color: #736e7d;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]: Can we derive this color from an existing variable?

File: src/components/ViewPage/ViewPage.css [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+  font-size: 17px;
+  line-height: 24px;
+  margin: 8px 0;
+  opacity: 0.8;
+  letter-spacing: -0.2px;
+  flex: 0 0 100%;
+}
+
+.ViewPage .detail .component-list {
+  color: #736e7d;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]: Can we derive this color from an existing variable?

File: src/components/Modals/ShareModal/ShareModal.css [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+  background: #fff;
+  border-radius: 8px 0 0 8px;
+  flex: 1 0;
+  border: 0;
+  font-size: 15px;
+  padding: 12px 0 12px 12px;
+}
+
+.ShareModal .share-modal .copy-group .copy-button {
+  background: #fff;
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.css]:
```suggestion
  background: white;
```

File: src/components/Modals/ShareModal/ShareModal.css [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+.ShareModal .share-modal .button-group,
+.ShareModal .share-modal .copy-group {
+  display: flex;
+  width: calc(480px - 48px - 48px);
+  justify-content: space-between;
+  margin-bottom: 12px;
+}
+
+.ShareModal .share-modal .copy-group .copy-input {
+  background: #fff;
```
Comment (belohlavek) [inline src/components/Modals/ShareModal/ShareModal.css]:
```suggestion
  background: white;
```

File: src/components/ViewPage/ViewPage.css [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+.ViewPage .detail .author .author-name {
+  color: #ff2d55;
+}
+
+.ViewPage .detail .avatar {
+  width: 24px;
+  height: 24px;
+  border-radius: 12px;  
+  margin-left: 12px;
+  background: #fff;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]:
```suggestion
  background: white;
```

File: src/components/ViewPage/ViewPage.css [hunk 24]
```diff
@@ -0,0 +1,121 @@
...
+
+.ViewPage .detail .author {
+  color: #736e7d;
+  font-size: 15px;
+  line-height: 24px;
+  margin: 0;
+}
+
+.ViewPage .detail .author .author-name {
+  color: #ff2d55;
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.css]: Is this --primary? can we derive from a variable?

File: src/components/ViewPage/ViewPage.tsx [hunk 25]
```diff
@@ -0,0 +1,161 @@
+import * as React from 'react'
+import { Props, State } from './ViewPage.types'
+import { Page, Loader } from 'decentraland-ui'
+// import { t } from 'decentraland-dapps/dist/modules/translation/utils'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]:
```suggestion
```

File: src/components/ViewPage/ViewPage.tsx [hunk 26]
```diff
@@ -0,0 +1,161 @@
+import * as React from 'react'
+import { Props, State } from './ViewPage.types'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]: Import order

File: src/components/ViewPage/ViewPage.tsx [hunk 27]
```diff
@@ -0,0 +1,161 @@
...
+import { Page, Loader } from 'decentraland-ui'
+// import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import NotFoundPage from 'components/NotFoundPage'
+import Footer from 'components/Footer'
+import Navbar from 'components/Navbar'
+import { Project } from 'modules/project/types'
+
+import './ViewPage.css'
+import ViewMenu from './ViewMenu'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]: Import order

File: src/components/ViewPage/ViewPage.tsx [hunk 28]
```diff
@@ -0,0 +1,161 @@
...
+    if (parcelCount === 0) {
+      return 0
+    }
+
+    let componentCount = 0
+    for (let component of Object.values(currentScene.components)) {
+      switch (component.type) {
+        case ComponentType.GLTFShape:
+        case ComponentType.NFTShape:
+          ++componentCount
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]:
```suggestion
          componentCount++
```

Just to follow the convention used elsewhere in the code 🙏

File: src/components/ViewPage/ViewPage.tsx [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
+import ViewMenu from './ViewMenu'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import Icon from 'components/Icon'
+import { ComponentType } from 'modules/scene/types'
+import ViewPort from 'components/ViewPort'
+import ShortcutTooltip from 'components/ShortcutTooltip'
+import { Shortcut } from 'modules/keyboard/types'
+import Chip from 'components/Chip'
+
+export default class ViewPage extends React.PureComponent<Props, State> {
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]: Can we think of a better name for this component? ViewPage is a bit too vague

File: src/components/ViewPage/ViewPage.tsx [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
+              <div style={{ flex: 1 }}/>
+              <div className="scene">
+                <ShortcutTooltip shortcut={Shortcut.PREVIEW} position="bottom center" className="tool" popupClassName="top-bar-popup">
+                  <Chip icon="preview" isActive={isPreviewing} onClick={this.handlePreview} />
+                </ShortcutTooltip>
+              </div>
+            </div>
+            <div className="thumbnail">
+              <ViewPort />
+              {/* <Loader active size="large" /> */}
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]:
```suggestion

```

File: src/components/ViewPage/ViewPage.tsx [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
+              {/* <img src={project.thumbnail} /> */}
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.tsx]:
```suggestion

```

File: src/components/ViewPage/ViewPage.types.ts [hunk 32]
```diff
@@ -0,0 +1,38 @@
+import { Dispatch } from 'redux'
+import { match } from 'react-router'
+
+// import { openModal, OpenModalAction } from 'modules/modal/actions'
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.types.ts]:
```suggestion
```

File: src/components/ViewPage/ViewPage.types.ts [hunk 33]
```diff
@@ -0,0 +1,38 @@
...
+  projectId: string,
+  type: 'public' | 'pool',
+  currentProject: Project | null
+  currentScene: Scene | null
+  currentAuthor: Profile | null
+  isPreviewing: boolean
+  isFetching: boolean
+  isLoggedIn: boolean
+  isReady: boolean
+  // onOpenModal: typeof openModal
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.types.ts]:
```suggestion
```

File: src/components/ViewPage/ViewPage.types.ts [hunk 34]
```diff
@@ -0,0 +1,38 @@
...
+
+export type State = {}
+
+export type MapStateProps = Pick<
+  Props,
+  'isPreviewing' | 'isFetching' | 'isLoggedIn' | 'isReady' | 'currentProject' | 'currentScene' | 'currentAuthor'
+>
+export type MapDispatchProps = Pick<Props, 'onLoadProject' | 'onPreview'>
+export type MapDispatch = Dispatch<
+  // | OpenModalAction
```
Comment (belohlavek) [inline src/components/ViewPage/ViewPage.types.ts]:
```suggestion
```

File: src/modules/profile/sagas.ts [hunk 35]
```diff
@@ -0,0 +1,20 @@
...
+  yield takeLatest(LOAD_PROFILE_REQUEST, handleLoadProfileRequest)
+}
+
+function* handleLoadProfileRequest(action: LoadProfileRequestAction) {
+  const { id } = action.payload
+  try {
+    const p: Profile = yield call(() => profile.fetchProfileById(id))
+    yield put(loadProfileSuccess(p))
+  } catch (e) {
+    console.log(e)
```
Comment (belohlavek) [inline src/modules/profile/sagas.ts]:
```suggestion
```

File: src/modules/profile/selectors.ts [hunk 36]
```diff
@@ -0,0 +1,21 @@
+import { createSelector } from 'reselect'
+
+import { RootState } from 'modules/common/types'
+import { Project } from 'modules/project/types'
+import { ProfileState } from './reducer'
```
Comment (belohlavek) [inline src/modules/profile/selectors.ts]: import order

File: src/modules/profile/types.ts [hunk 37]
```diff
@@ -0,0 +1,2 @@
+import { User } from "modules/auth/types"
+export type Profile = User
```
Comment (belohlavek) [inline src/modules/profile/types.ts]: Why not use User directly?

File: src/modules/profile/utils.ts [hunk 38]
```diff
@@ -0,0 +1,88 @@
+import { Project } from 'modules/project/types'
```
Comment (belohlavek) [inline src/modules/profile/utils.ts]: Is this utils necessary?

File: src/modules/ui/share/actions.ts [hunk 39]
```diff
@@ -0,0 +1,6 @@
+import { action } from 'typesafe-actions'
+import { ShareTarget } from './types'
+
+export const SHARE = 'Share'
```
Comment (belohlavek) [inline src/modules/ui/share/actions.ts]: Let's rename this to SHARE_SCENE or something similar

File: src/routing/Routes.tsx [hunk 40]
```diff
@@ -15,6 +15,7 @@ import UnsupportedBrowserPage from 'components/UnsupportedBrowserPage'
 import LoadingPage from 'components/LoadingPage'
 
 import { Props, State } from './Routes.types'
+import ViewPage from 'components/ViewPage'
```
Comment (belohlavek) [inline src/routing/Routes.tsx]: Import order

**Comment by 2fd** [comment]
@belohlavek ready!

**Pull request merged**